contracts/TaskBook.sol

```diff
@@ -0,0 +1,89 @@
+pragma solidity ^0.4.4;
+
+import './AccountManager.sol';
+
+contract TaskBook is AccountManager {
+
+	uint private numTasks = 0;
+	uint constant maxSolvers = 10;
+	uint constant maxChallengers = 10;
+
+	event NewTask(uint taskID, uint minDeposit, uint blockNumber);
+	event SolveTask(uint indexed taskID, address solver, bytes32 taskData, uint minDeposit);
+	event SubmitSolution(uint taskID, bytes32 solution, uint minDeposit, bytes32 taskData);
+
+	mapping(uint => Task) private tasks;
+	mapping(uint => bytes32) private solverRandomBits;
+	mapping(uint => mapping(address => uint)) private challengers;
+
+	struct Task {
+		address owner;
+		address[] solvers;
+		address selectedSolver;
+		uint minDeposit;
+		bytes32 taskData;
+		uint numSolvers;
+		address[] challengers;
+		uint numChallengers;
+	}
+
+	//Task Issuers create tasks to be solved
+	function newTask(uint minDeposit, bytes32 taskData, uint numBlocks) returns (bool) {
+		require(balances[msg.sender] >= minDeposit);
+		tasks[numTasks] = Task(msg.sender, new address[](maxSolvers), 0x0, minDeposit, taskData, 0, new address[](maxSolvers), 0);
+		log0(sha3(msg.sender));//possible bug if log is after event
+		NewTask(numTasks, minDeposit, block.number+numBlocks);
+		numTasks++;
+		return true;
+	}
+
+	//Solver registers for tasks
+	function registerForTask(uint id, uint minDeposit) returns(bool) {
+		require(balances[msg.sender] >= minDeposit);
+		Task t = tasks[id];
+		require(!(t.owner == 0x0));
```

Comment:
nitpick: is `!= 0x0` more explicit?

Reply:
Not sure, was going for a check to see if it was "null"